Applications/Classification/README.md

```diff
@@ -1,3 +1,89 @@
-# Classifciation
+# Classification
 
+This example demonstrate the transfer learning with mobilenetv2 which pre-trained for imagenet. "mobilenetv2.tflite" is used for feature extractor which is removed last fully connected layer manually.
 
+After feature extraction, it is used for training fully connencted layer. To take the feature as an input, we add input layer before fully connected layer.
+
+Total number of data for training is 10 (number of class) * 100 (number of image for each class)
+
+This example has two options to train.
+### 1. Train with training set files ( main.cpp )
+   - This option is activated if [ DataSet ] section is configured in configuration file.
+   - This assumes that there are training set which saved as an file.
+     For ith image, feature data (62720 x sizeof(float)) + label (10 x sizeof(float)) at i x ((62720 x sizeof(float) + label(10 x sizeof(float)) position.
+   - The input image (bmp format : 32x32x3) is stored as in ExtractFeatures() at "data_path" directory defined as argv #1.
+   ```
+     string total_label[10] = {"airplane", "automobile", "bird", "cat", "deer",
+                            "dog", "frog", "horse", "ship", "truck"};
+```
+   - Training dataset files are generated 'getMiniBatch()' function in main.cpp.
+   - Once it generates the dataset files, it start to read and feed the buffer to train.
+
+### 2. Train with training data generator ( main_func.cpp )
+   - This option is activated if there is no [ DataSet ] section is defined.
+   - Expect to get the generator function whcih defined by user.
+   - This option just use training data set file which is generated by option 1.
+   - training data set generator is getMiniBatch_train(),  getMiniBatch_val() for validation. test data set is not used.
+
+Classification.ini file (res directory) is input which has configuration and hyper-parameters of network for nntrainer.
+
+## How to run examples
+
+### Preparing NNTrainer
+* https://github.com/nnstreamer/nntrainer/blob/master/docs/getting-started.md
+
+### Write Configuration file
+
+you can find detail explanation about each keyward in
+https://github.com/nnstreamer/nntrainer/blob/master/docs/configuration-ini.md
+
+```
+[Network]
+Type = NeuralNetwork
+Learning_rate = 0.0001
+Decay_rate = 0.96
+Decay_steps = 1000
+Epoch = 30000
+Optimizer = adam
+Cost = cross
+Model = "model.bin"
+minibatch = 32
+beta1 = 0.9
+beta2 = 0.9999
+epsilon = 1e-7
+
+[DataSet]
+BufferSize=100
+TrainData="trainingSet.dat"
+ValidData="trainingSet.dat"
+LabelData="label.dat"
+
+[inputlayer]
+Type = input
+Input_Shape = 1:1:62720
+Bias_init_zero = true
+Normalization = true
+
+[outputlayer]
+Type = fully_connected
+Unit = 10
+Bias_init_zero = true
+Activation = softmax
+Weight_Decay = l2norm
+weight_Decay_Lambda = 0.005
+```
+
+If you want to use generator (option #2), then remove [DataSet] section. Then it expect to have generator function automatically.
```

Comment:
If you want to use generator (option #2), then remove [DataSet] section, and provide dataset generator callbacks.